File: admin-tools/profile_rec.py


class ProfileRec: 
    
    PENDING = "pending"
    INVITED = "invited"
    DECLINED = "declined"
    INVITED_DECLINED = "invitedDeclined"
    INVITED_ACCEPTED = "invitedAccepted"
    
    def __init__(self, status = PENDING, profileViews = 0): 
        self.status = status
        self.profileViews = profileViews
    
    
    def to_dict(self):
        return {
            "status": self.status,
            "profileViews": self.profileViews,
            "added_day": self.added_day,
            "acted_at": self.acted_at,
        }

File: Data/Models/Profile/ProfileRec.swift
//
//  ProfileRecModel.swift
//  Scoop
//
//  Created by Art Ostin on 14/02/2026.
//

import Foundation
@preconcurrency import FirebaseFirestore

//Data type stored in Firebase for Profile Recommendations
struct ProfileRec: Identifiable, Codable, Sendable{
    
    enum Field: String {
        case id, profileViews, status, addedDay, updatedDay
    }
    
    enum Status: String, Codable, Sendable {
        case pending, invited, declined, invitedDeclined, invitedAccepted
    }
    
    @DocumentID var id: String?
    var profileViews: Int
    var status: Status
    @ServerTimestamp var addedDay: Timestamp?
    var updatedDay: Timestamp?
}
